src/app/auth/auth.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { SignupCredentials } from '../signup_credentials';
+import { SigninCredentials } from '../signin_credentials';
+import { BehaviorSubject, tap, Observable } from 'rxjs';
+
+interface SignupResponse {
+  username: string;
+}
+
+interface SignedinResponse {
+  authenticated: boolean;
+  username: string;
+}
+
+interface SigninResponse {
+  username: string;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  rootUrl: string = 'https://api.angular-email.com';
+  signedin$: BehaviorSubject<any> = new BehaviorSubject(null);
```

Comment:
Add type casting for everything. Using `any` should be avoided as much as possible. Especially in this case where there are interfaces already defined for this response.